### ISSUE TITLE
error installing due to 'dezalgo' package

### ISSUE THREAD

**Issue by ShawnTalbert** [aurelia/cli#809]
**I'm submitting a bug report**
* **Library Version:**
0.32.0

**Please tell us about your environment:**
* **Operating System:**
Linux (Mint 18)

* **Node Version:**
8.9.4
* **NPM Version:**
5.6.0
**Current behavior:**
Installing via npm locally (no `-g`) the package fails. I'm installing with `npm install aurelia-cli`

```
1019 verbose stack Error: ENOENT: no such file or directory, rename '/home/shawn/node_modules/aurelia-cli/node_modules/npm/node_modules/dezalgo' -> '/home/shawn/node_modules/aurelia-cli/node_modules/npm/node_modules/.dezalgo.DELETE'
1020 verbose cwd /home/shawn
1021 verbose Linux 4.10.0-42-generic
1022 verbose argv "/home/shawn/node-v8.9.4-linux-x64/bin/node" "/home/shawn/bin/npm" "install" "aurelia-cli"
1023 verbose node v8.9.4
1024 verbose npm  v5.6.0
1025 error path /home/shawn/node_modules/aurelia-cli/node_modules/npm/node_modules/dezalgo
1026 error code ENOENT
1027 error errno -2
1028 error syscall rename
1029 error enoent ENOENT: no such file or directory, rename '/home/shawn/node_modules/aurelia-cli/node_modules/npm/node_modules/dezalgo' -> '/home/shawn/node_modules/aurelia-cli/node_modules/npm/node_modules/.dezalgo.DELETE'
1030 error enoent This is related to npm not being able to find a file.
1031 verbose exit [ -2, true ]
```

As a workaround, if I first install dezalgo (e.g. `npm install dezalgo`) then install the cli it works.

**Comment by JeroenVinke** [comment]
Hmm this looks to be an npm issue: https://github.com/npm/npm/issues/17444

**Comment by AshleyGrant** [comment]
Yep, downgrading to npm v5.5.1 fixes the issue for me for now. I've confirmed this across multiple developers over the last two weeks to work as a temporary fix.

**Comment by nenadvicentic** [comment]
I've ran into the same issue, but to me it happens only when `node_modules` folder contains `aurelia-cli` package.

I was testing this on already existing Aurelia project, already containing needed npm packages in `node_modules`.

1. `npm install <any package>` fails with above described error (`.dezalgo.DELETE`).
2. remove Aureli CLI by running `npm remove aurelia-cli`
3. `npm install <any package>` works again (but obviously cannot build the project without local CLI package).
4. You can try running `npm install aurelia-cli', after you have installed all additional packages.
5. But now `au build` complains that it cannot find `gulp`.

My impression is that problem is interaction/collision between `npm` inside of `aurelia-cli` and global `npm`. Basically, first `npm install` when `node_modules` folder is empty will always work, but as soon as `aurelia-cli` is unpacked in `node_modules`, `.dezalgo.DELETE` error pops up.

**Comment by Alexander-Taran** [comment]
world is moving towards yarn

**Comment by fkleuver** [comment]
This is directly related to the way package-lock.json is resolved since npm@5.6.0. For aurelia-cli it happens because of the dependency on `npm@3.10.8` as somewhat clarified [here](https://github.com/npm/npm/issues/17444#issuecomment-365692331) and [here](https://github.com/npm/npm/issues/19304#issuecomment-358384940).

### Possible workarounds (pick any):

- Downgrade global npm version to 5.5.1
- Delete package-lock.json and/or node_modules folder prior to `npm install`
- Upgrade `aurelia-cli`'s dependency on `npm` to 5.0.0 or above

As for the third option; I actually forked aurelia-cli and upgraded it's dependency on npm to 5.6.0, did the `npm link` dance and I just use `npm link` whenever I need aurelia-cli as a dependency. The problem hasn't occurred for me since, through multiple projects.

Is there any particular reason aurelia-cli depends on `npm@^3.10.8`? Otherwise why not just upgrade that dependency and be done with this silly issue?

**Comment by omniscient** [comment]
I agree with @fkleuver. Team, can we just move the cli npm dependency to v5.6 and be done with it?

**Comment by AshleyGrant** [comment]
@JeroenVinke let's get the dep upgraded and get the issue closed.

**Comment by JeroenVinke** [comment]
Yeah I agree

**Comment by Alexander-Taran** [comment]
https://github.com/npm/npm/issues/19883
and
https://www.google.ru/search?newwindow=1&ei=LjCXWr2AM5DCwQKSlovADA&q=npm+5.7.0++production+servers&oq=npm+5.7.0++production+servers&gs_l=psy-ab.3...58763.62414.0.62684.7.7.0.0.0.0.90.570.7.7.0....0...1c.1.64.psy-ab..0.0.0....0.HnE2QhgtQLQ

for that matter..

please don't rush

**Comment by AshleyGrant** [comment]
That's v5.7.0, which isn't the current release. v5.6.0 is the current release, and I don't believe it has the terrible bug in it.

**Comment by Alexander-Taran** [comment]
<3 yarn

**Comment by maroy1986** [comment]
Yeah that was only in 5.7.0 which is a pre release version. 5.6.0 is fine.

If we really want to avoid trouble with future release, why not locking to 5.6.0 for now in the CLI package.json since we know it's working fine? Anyway, we were kind of locked on the latest 3.x version...

**Comment by AshleyGrant** [comment]
Closed via #832
